tasks.go

```diff
@@ -117,15 +117,18 @@ func NewError(err error) Task {
 }
 
 // GetNagiosTasks sends tasks to dump Nagios data for each project that contain
-// a Nagios pod.
-func GetNagiosTasks(tasks chan<- Task, projects []string, basepath string) {
+// a Nagios pod. This function will output an error to the user if no Nagios pods
+// were found in any projects.
+func GetNagiosTasks(tasks chan<- Task, projects []string, basepath string, resourceFactory ResourceMatchFactory) {
+	foundANagiosPod := false
 	for _, p := range projects {
-		pods, err := getResourceNamesBySubstr(p, "pod", "nagios")
+		pods, err := resourceFactory(p, "pod", "nagios")
 		if err != nil {
 			tasks <- NewError(err)
 			continue
 		}
 		for _, pod := range pods {
+			foundANagiosPod = true
 			outFor := outToFile(basepath, "dat", "nagios")
```

Comment:
I don't see an easy way to test the inside of this for loop without passing the outToFile function so it can be mocked in the test. Is it worth doing that now?